File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/data/model/Expense.java
package com.github.arhor.simple.expense.tracker.data.model;

import lombok.Builder;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Immutable;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Table;

@Table("expenses")
@Builder(toBuilder = true)
@Immutable
public record Expense(
    @Id
    Long id,
    Long userId,
    String name,
    String icon,
    String color,
    @CreatedDate
    LocalDateTime created,
    @LastModifiedDate
    LocalDateTime updated
) {
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/data/model/Expense.kt
package com.github.arhor.simple.expense.tracker.data.model

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Immutable
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Immutable
@Table(name = "expenses")
data class Expense(
    @Id
    val id: Long? = null,
    val userId: Long,
    val name: String,
    val icon: String?,
    val color: String?,
    @CreatedDate
    val created: LocalDateTime? = null,
    @LastModifiedDate
    val updated: LocalDateTime? = null,
)
